gr-blocks/lib/correctiq_man_impl.cc

```diff
@@ -0,0 +1,127 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "correctiq_man_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace blocks {
+
+correctiq_man::sptr correctiq_man::make(float real, float imag)
+{
+    return gnuradio::get_initial_sptr(new correctiq_man_impl(real, imag));
+}
+
+/*
+ * The private constructor
+ */
+correctiq_man_impl::correctiq_man_impl(float real, float imag)
+    : gr::sync_block("correctiq_man",
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)),
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)))
+{
+    avg_real = real;
+    avg_img = imag;
+    d_k = gr_complex(avg_real, avg_img);
+
+    message_port_register_in(pmt::mp("set_real"));
+    set_msg_handler(pmt::mp("set_real"),
+                    [this](pmt::pmt_t msg) { this->handle_real(msg); });
+    message_port_register_in(pmt::mp("set_imag"));
+    set_msg_handler(pmt::mp("set_imag"),
+                    [this](pmt::pmt_t msg) { this->handle_imag(msg); });
+}
+
+/*
+ * Our virtual destructor.
+ */
+correctiq_man_impl::~correctiq_man_impl() {}
+
+float correctiq_man_impl::get_real() { return avg_real; }
+float correctiq_man_impl::get_imag() { return avg_img; }
+
+void correctiq_man_impl::set_real(float new_value)
+{
+    avg_real = new_value;
+    d_k = gr_complex(avg_real, avg_img);
+}
+
+void correctiq_man_impl::set_imag(float new_value)
+{
+    avg_img = new_value;
+    d_k = gr_complex(avg_real, avg_img);
+}
+
+void correctiq_man_impl::handle_real(pmt::pmt_t msg)
+{
+    if (pmt::is_pair(msg)) {
+        pmt::pmt_t data = pmt::cdr(msg);
+        if (pmt::is_real(data)) {
+            float new_value = pmt::to_float(data);
+
+            set_real(new_value);
+        } else {
+            GR_LOG_INFO(d_logger, "ERROR: non-float real value received.");
```

Comment:
you could use `GR_LOG_ERROR` instead of `_INFO` here :) but this is not a clear-cut case. This might really be an error, or you foresee an application where this might "just happen", in which case `_INFO` would be right.